File: Frontend/untitled/src/app/services/product-image.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface ProductImageDTO {
  id?: string;
  imageData?: string;
  contentType?: string;
  isPrimary?: boolean;
  productId?: string;
}

@Injectable({ providedIn: 'root' })
export class ProductImageService {
  private apiUrl = '/api/product-images';

  constructor(private http: HttpClient) {}

  findByProductId(productId: string): Observable<ProductImageDTO[]> {
    return this.http.get<ProductImageDTO[]>(`${this.apiUrl}/product/${productId}`);
  }

  create(dto: ProductImageDTO): Observable<any> {
    return this.http.post(this.apiUrl, dto);
  }

  setPrimary(id: string): Observable<any> {
    return this.http.patch(`${this.apiUrl}/${id}/primary`, {});
  }

  deleteBatch(ids: string[]): Observable<any> {
    return this.http.delete(`${this.apiUrl}/batch`, { body: ids });
  }
}

File: Backend/src/main/java/com/example/demo/service/ProductImageService.java
package com.example.demo.service;

import com.example.demo.dto.ProductImageDTO;
import com.example.demo.model.ProductImage;

import java.util.List;

public interface ProductImageService {
    void create(ProductImageDTO image);

    void delete(ProductImageDTO image);

    ProductImageDTO findById(String id);

    List<ProductImageDTO> findByProductId(String productId);

    void deleteByProductId(String productId);

    void createBatch(List<ProductImageDTO> images);

    void deleteImagesByIds(List<String> ids);
    void setPrimary(String imageId);
}
